### PR TITLE
Fix Issue 24125 - ImportC: vector type initializer not understood

### PR DESCRIPTION
**Pull request by WalterBright**
It took a while to figure out just where the adjustments needed to go, so while it's a small number of lines changes I wouldn't say it was an easy review.

But it does generate the same code as the equivalent D version.

### CODE REVIEW

File: compiler/src/dmd/cparse.d [hunk 1]
```diff
@@ -1890,6 +1890,14 @@ final class CParser(AST) : Parser!AST
                 if (specifier.alignExps)
                     error("no alignment-specifier for typedef declaration"); // C11 6.7.5-2
 
+                if (specifier.vector_size)
+                {
+                    auto length = new AST.IntegerExp(token.loc, specifier.vector_size / dt.size(), AST.Type.tuns32);
+                    auto tsa = new AST.TypeSArray(dt, length);
+                    dt = new AST.TypeVector(tsa);
+                    specifier.vector_size = 0;          // used it up
```
Comment (dkorpel) [inline compiler/src/dmd/cparse.d:1898]: Why is this needed?

**Review by dkorpel — APPROVED** [review]

File: compiler/src/dmd/cparse.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/cparse.d:1898]: To fix the first problem mentioned in https://issues.dlang.org/show_bug.cgi?id=24125#c2
